While implementing the suggestions, discovered that I flubbed testing the previous update (to use the PyPi library [python-simple-hipchat-v2](https://github.com/CDKGlobal/python-simple-hipchat-v2)) and didn't actually test that version (as in tested sending to hipchat, it passed Tox), but my previous version.

Apparently, it doesn't actually work:
```python
Traceback (most recent call last):
  File "uvloop/future.pyx", line 374, in uvloop.loop.BaseTask._fast_step (uvloop/loop.c:112704)
  File "/usr/src/app/homeassistant/core.py", line 1015, in _event_to_service_call
    yield from service_handler.func(service_call)
  File "/usr/src/app/homeassistant/components/notify/__init__.py", line 136, in async_notify_message
    yield from notify_service.async_send_message(**kwargs)
  File "uvloop/future.pyx", line 230, in __iter__ (uvloop/loop.c:110600)
  File "uvloop/future.pyx", line 432, in uvloop.loop.BaseTask._fast_wakeup (uvloop/loop.c:113980)
  File "uvloop/future.pyx", line 101, in uvloop.loop.BaseFuture._result_impl (uvloop/loop.c:108900)
  File "/usr/local/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/notify/hipchat.py", line 97, in send_message
    message_format=message_format)
  File "/config/deps/hipchat_v2/__init__.py", line 52, in message_room
    return self.method(url, headers=headers, data=data)
  File "/config/deps/hipchat_v2/__init__.py", line 35, in method
    response = self.opener.open(req, None, timeout).read()
  File "/usr/local/lib/python3.5/urllib/request.py", line 464, in open
    req = meth(req)
  File "/usr/local/lib/python3.5/urllib/request.py", line 1183, in do_request_
    raise TypeError(msg)
TypeError: POST data should be bytes or an iterable of bytes. It cannot be of type str.
```

Supposedly this library works on Python 2 & 3. Looking at the code for it, it passes the output of json.dumps which is a string to six.moves.urllib.request.Request. Supposedly six implements some functionality to make things work on python 2 & 3, but something somewhere doesn't work obviously, as it errors out expecting bytes not string.

I guess I'll switch to the other PyPi library [hipnotify](https://github.com/achiku/hipnotify) and see if it works. Downside is that I'll have to either drop allowing room selection on the call to notify or implement some extra logic to create and manage Room objects on-demand since this library sets the room at initialization, not during the call to the send message function.

--- a/homeassistant/components/notify/hipchat.py
+++ b/homeassistant/components/notify/hipchat.py
@@ -57,10 +57,7 @@ class HipchatNotificationService(BaseNotificationService):
         """Initialize the service."""
         self._token = token
         self._default_room = default_room
-        if default_color in VALID_COLORS:
-            self._default_color = default_color
-        else:
-            self._default_color = 'yellow'
+        self._default_color = default_color
         self._default_notify = default_notify
         self._default_format = default_format
         self._host = host

--- a/homeassistant/components/notify/hipchat.py
+++ b/homeassistant/components/notify/hipchat.py
@@ -15,7 +15,7 @@ from homeassistant.const import (
     CONF_TOKEN)
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['python-simple-hipchat-v2==0.0.1']
+REQUIREMENTS = ['hipnotify==1.0.8']
 
 _LOGGER = logging.getLogger(__name__)
 
@@ -25,12 +25,15 @@ CONF_NOTIFY = 'notify'
 CONF_FORMAT = 'format'
 CONF_HOST = 'host'
 
+VALID_COLORS = {'yellow', 'green', 'red', 'purple', 'gray', 'random'}
+VALID_FORMATS = {'text', 'html'}
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_TOKEN): cv.string,
-    vol.Required(CONF_ROOM): cv.string,
-    vol.Optional(CONF_COLOR, default='yellow'): cv.string,
+    vol.Required(CONF_ROOM): vol.Coerce(int),
+    vol.Optional(CONF_COLOR, default='yellow'): vol.In(VALID_COLORS),
     vol.Optional(CONF_NOTIFY, default=False): cv.boolean,
-    vol.Optional(CONF_FORMAT, default='text'): cv.string,
+    vol.Optional(CONF_FORMAT, default='text'): vol.In(VALID_FORMATS),
     vol.Optional(CONF_HOST, default='https://api.hipchat.com/'): cv.string,
 })
 
@@ -49,27 +52,30 @@ def get_service(hass, config, discovery_info=None):
 class HipchatNotificationService(BaseNotificationService):
     """Implement the notification service for HipChat."""
 
-    _valid_colors = {'yellow', 'green', 'red', 'purple', 'gray', 'random'}
-    _valid_formats = {'text', 'html'}
-
     def __init__(self, token, default_room, default_color, default_notify,
                  default_format, host):
         """Initialize the service."""
-        import hipchat_v2
         self._token = token
         self._default_room = default_room
-        if default_color in self._valid_colors:
+        if default_color in VALID_COLORS:
             self._default_color = default_color
         else:
             self._default_color = 'yellow'
-        if isinstance(default_notify, bool):
-            self._default_notify = default_notify
-        else:
-            self._default_notify = False
+        self._default_notify = default_notify
         self._default_format = default_format
         self._host = host
 
-        self._hipchat = hipchat_v2.HipChat(token=self._token, url=self._host)
+        self._rooms = {}
+        self._get_room(self._default_room)
+
+    def _get_room(self, room):
+        """Get Room object, creating it if necessary."""
+        from hipnotify import Room
+        if room not in self._rooms:
+            self._rooms[room] = Room(token=self._token,
+                                     room_id=room,
+                                     endpoint_url=self._host)
+        return self._rooms[room]
 
     def send_message(self, message="", **kwargs):
         """Send a message."""
@@ -80,24 +86,20 @@ class HipchatNotificationService(BaseNotificationService):
         if kwargs.get(ATTR_DATA) is not None:
             data = kwargs.get(ATTR_DATA)
             if ((data.get(CONF_COLOR) is not None)
-                    and (data.get(CONF_COLOR) in self._valid_colors)):
+                    and (data.get(CONF_COLOR) in VALID_COLORS)):
                 color = data.get(CONF_COLOR)
             if ((data.get(CONF_NOTIFY) is not None)
                     and isinstance(data.get(CONF_NOTIFY), bool)):
                 notify = data.get(CONF_NOTIFY)
             if ((data.get(CONF_FORMAT) is not None)
-                    and (data.get(CONF_FORMAT) in self._valid_formats)):
+                    and (data.get(CONF_FORMAT) in VALID_FORMATS)):
                 message_format = data.get(CONF_FORMAT)
 
-        if kwargs.get(ATTR_TARGET) is None:
-            targets = [self._default_room]
-        else:
-            targets = kwargs.get(ATTR_TARGET)
+        targets = kwargs.get(ATTR_TARGET, [self._default_room])
 
-        for room in targets:
-
-            self._hipchat.message_room(room_id=room,
-                                       message=message,
-                                       color=color,
-                                       notify=notify,
-                                       message_format=message_format)
+        for target in targets:
+            room = self._get_room(target)
+            room.notify(msg=message,
+                        color=color,
+                        notify=notify,
+                        message_format=message_format)

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -283,6 +283,9 @@ heatmiserV3==0.9.1
 # homeassistant.components.switch.hikvisioncam
 hikvision==0.4
 
+# homeassistant.components.notify.hipchat
+hipnotify==1.0.8
+
 # homeassistant.components.binary_sensor.workday
 holidays==0.8.1
 
@@ -755,9 +758,6 @@ python-ripple-api==0.0.2
 
 # homeassistant.components.media_player.roku
 python-roku==3.1.3
-
-# homeassistant.components.notify.hipchat
-python-simple-hipchat-v2==0.0.1
 
 # homeassistant.components.sensor.synologydsm
 python-synology==0.1.0

